spring-cloud-kubernetes-client-config/src/main/java/org/springframework/cloud/kubernetes/client/config/KubernetesClientConfigMapPropertySource.java

```diff
@@ -39,18 +39,25 @@
 
 	private static final Log LOG = LogFactory.getLog(KubernetesClientConfigMapPropertySource.class);
 
+	@Deprecated
 	public KubernetesClientConfigMapPropertySource(CoreV1Api coreV1Api, String name, String namespace,
-			Environment environment, String prefix) {
-		super(getName(name, namespace), getData(coreV1Api, name, namespace, environment, prefix));
+			Environment environment) {
+		super(getName(name, namespace), getData(coreV1Api, name, namespace, environment, "", true));
+	}
+
+	public KubernetesClientConfigMapPropertySource(CoreV1Api coreV1Api, String name, String namespace,
+			Environment environment, String prefix, boolean useProfileNameAsSuffix) {
+		super(getName(name, namespace),
+				getData(coreV1Api, name, namespace, environment, prefix, useProfileNameAsSuffix));
 	}
 
 	private static Map<String, Object> getData(CoreV1Api coreV1Api, String name, String namespace,
-			Environment environment, String prefix) {
+			Environment environment, String prefix, boolean useProfileNameAsSuffix) {
```

Comment:
this is the crox of the change, pretty much.

Reply:
I don't really understand why you would not want to pick up config maps and secrets with the profile name.  This is how Spring Boot has always worked with configuration properties so it seems like that is a fair assumption.

Reply:
this is not about my preference, really. We are fixing [this](https://github.com/spring-cloud/spring-cloud-kubernetes/issues/508)